authheader.go

```diff
@@ -5,26 +5,46 @@ import (
 	"strings"
 )
 
-type authheader string
+type authheader []string
 
-func (h authheader) IsBasic() bool {
-	return strings.HasPrefix(string(h), "Basic ")
+func (h authheader) IsBasic() (bool, string) {
```

Comment:
Kind of sad to see this one have a different signature than the other `IsXXX()` methods. Maybe let this return `bool` only and add `func (h authheader) Basic() (string)` to return the auth header that is prefixed with basic? This method could then just call Basic() and check for empty string?

Reply:
Thanks @paulmey for these comments, I've just did the following: Create of Basic() function next to the IsBasic()